src/main/java/com/faforever/client/replay/ReplayFileReaderImpl.java

```diff
@@ -34,16 +37,22 @@ public ReplayFileReaderImpl() {
   @SneakyThrows
   public LocalReplayInfo parseMetaData(Path replayFile) {
     logger.debug("Parsing metadata of replay file: {}", replayFile);
-    List<String> lines = Files.readAllLines(replayFile);
-    return gson.fromJson(lines.get(0), LocalReplayInfo.class);
+    Optional<String> metadata = Files.lines(replayFile).findFirst();
+    if (metadata.isPresent()) {
+      return gson.fromJson(metadata.get(), LocalReplayInfo.class);
+    }
+    throw new IOException(String.format("Failed to extract metadata from replay file: {}", replayFile));
   }
 
   @Override
   @SneakyThrows
   public byte[] readRawReplayData(Path replayFile) {
     logger.debug("Reading replay file: {}", replayFile);
-    List<String> lines = Files.readAllLines(replayFile);
-    return QtCompress.qUncompress(BaseEncoding.base64().decode(lines.get(1)));
+    Optional<String> replayData = Files.lines(replayFile).skip(1).findFirst();
+    if (replayData.isPresent()) {
+      return QtCompress.qUncompress(BaseEncoding.base64().decode(replayData.get()));
```

Comment:
Does this `QtCompres.qUncompress` decompress the stuff right?
https://github.com/norraxx/faf-scfa-replay-api/blob/master/source/utils.py#L21

Do we need there shift by 4 bytes?

Reply:
Hm. The previous code didn't do that either. I suspect the 4 bytes are the linebreak (CRLF).